commands/audit/sca/pnpm/pnpm.go

```diff
@@ -46,10 +48,22 @@ func BuildDependencyTree(params utils.AuditParams) (dependencyTrees []*xrayUtils
 		return
 	}
 	// Build
-	if err = installProjectIfNeeded(pnpmExecPath, currentDir); errorutils.CheckError(err) != nil {
+	var tempDirForDependenciesCalculation string
+	if tempDirForDependenciesCalculation, err = installProjectIfNeeded(pnpmExecPath, currentDir); errorutils.CheckError(err) != nil {
 		return
 	}
-	return calculateDependencies(pnpmExecPath, currentDir, params)
+
+	var dirToCalcDependenciesOn string
+	if tempDirForDependenciesCalculation == "" {
+		dirToCalcDependenciesOn = currentDir
+	} else {
+		// If tempDirForDependenciesCalculation contains a non-empty value, it means we executed 'install' on a temporary directory on which we need to calculate the dependencies and remove it at the end
+		dirToCalcDependenciesOn = tempDirForDependenciesCalculation
+		defer func() {
+			err = errors.Join(err, biutils.RemoveTempDir(tempDirForDependenciesCalculation))
+		}()
+	}
+	return calculateDependencies(pnpmExecPath, dirToCalcDependenciesOn, params)
 }
```

Comment:
I think it is better to install the project in an isolated directory anyway. Those conditions make the code harder to read, harder to maintain, and harder to extend. It's a better practice not to work in their project directory.